test/DurableTask.Netherite.Tests/FaultInjectionTests.cs

```diff
@@ -64,6 +64,42 @@ public async Task InjectStartup()
             fixture?.Dispose();
         }
 
+
+        [Fact]
+        public async Task CanRecoverFromFailedStartup()
+        {
+            HostFixture fixture = null;
+
+            // first force a permnanent failure
+            using (this.faultInjector.WithMode(Faster.FaultInjector.InjectionMode.PermanentFail, injectDuringStartup: true))
+            {
+                try
+                {
+                    fixture = new HostFixture(this.settings, useCacheDebugger: true, useReplayChecker: true, restrictMemory: null, output: (msg) => this.outputHelper.WriteLine(msg));
+                    await fixture.Host.StartAsync();
+                }
```

Comment:
the test should check here that an exception is actually thrown.

But I think the best way to do this is actually to just use the `Assert.ThrowsAsync` construct instead of a try clause:

```csharp
await Assert.ThrowsAsync<Exception>(async () =>
{
    ...
});
```

Reply:
good point, I tend to forget this utility exists. I've refactored this accordingly